packages/wallet/src/redux/reducer.ts

```diff
@@ -51,6 +52,11 @@ export function initializedReducer(
     newState = updateSharedData(newState, action);
   }
 
+  // if the action is both a protocol action and a new process action (e.g. action terminates current process and starts new one)
+  if (isNewProcessAction(action) && actions.isProtocolAction(action)) {
```

Comment:
Instead of having an `action` that creates a new process and terminates an old one would it make sense to just have two actions? IE (`ExitChallenge` and `DefundRequested`).  That way we can keep the distinction between new process actions and protocol actions.

Reply:
In principle yes, but that would seem to steer us toward an internal message queue (something that we may not wish to have).

Reply:
If (in future work) these protocols are nested under `Application-`, they won't need to be spawned as new processes, and this action can just be a `ProtocolAction`.